lib/SILOptimizer/Utils/InstOptUtils.cpp

```diff
@@ -1963,3 +1963,45 @@ SILBasicBlock::iterator swift::replaceSingleUse(Operand *use, SILValue newValue,
 
   return nextII;
 }
+
+SILValue swift::makeValueAvailableForOwnedUse(
+    SILValue value, SILBasicBlock *useBB,
+    JointPostDominanceSetComputer *jointPostDomComputer) {
+  if (!value->getFunction()->hasOwnership())
+    return value;
+
+  if (value->getType().isTrivial(*value->getFunction()))
+    return value;
+
+  auto insertPt = getInsertAfterPoint(value).getValue();
+
+  // If \p value is a newly created @owned value, we don't have to create a copy
```

Comment:
This check doesn't make sense: as this utility function is a general purpose utility, it should assume that the SIL is valid when calling it. An owned value without any uses is not valid SIL.
If you actually hit this case, I would be concerned.

That means: you should always copy the value.
Maybe a better name for the function would then be `makeCopiedValueAvailable` (the "...forOwnedUse" is implied by "Copied").

Reply:
This check was needed for newly created owned values passed into the utility. 
In RLE, for example, if we have struct field loads followed by a struct load, RLE creates a struct (similarly for tuple) to be used instead of the redundant load. This case, where we have a newly created struct does not need an additional copy. (May need a control equivalent copy).

Reply:
@eeckstein I updated the PR with 2 versions of the api `makeCopiedValueAvailable`  and `makeNewValueAvailable` (for new values without uses) now.